### PR TITLE
feat: making appData immutable

### PR DESCRIPTION
**Pull request by wei3erHase**
Closes BAL-112

### CODE REVIEW

File: script/Params.s.sol [hunk 1]
```diff
@@ -9,24 +9,28 @@ contract Params {
...
     address settlement;
+    bytes32 appData;
   }
 
+  /// @notice Settlement address
+  address internal constant _GPV2_SETTLEMENT = 0x9008D19f58AAbD9eD0D60971565AA8510560ab41;
+
+  /// @notice AppData identifier
+  bytes32 internal constant _APP_DATA = bytes32('appData');
+
```
Comment (wei3erHase) [inline script/Params.s.sol:20]: fixing linter warning about constants order declaration (moving up constants)

File: test/integration/BCowPool.t.sol [hunk 2]
```diff
@@ -26,13 +26,10 @@ contract BCowPoolIntegrationTest is PoolSwapIntegrationTest, BCoWConst {
 
   function setUp() public override {
     super.setUp();
-
-    // enable trading
-    IBCoWPool(address(pool)).enableTrading(APP_DATA);
   }
```
Comment (0xAustrian) [inline test/integration/BCowPool.t.sol:29]: Then we dont need setup in this contract

**Review by 0xAustrian — COMMENTED** [review]
Looks good, remember to remove `NO_TRADING` constant from `BCoWConst.sol`

File: src/interfaces/IBCoWPool.sol [hunk 3]
```diff
@@ -65,29 +51,16 @@ interface IBCoWPool is IERC1271, IBPool {
...
 
   /**
-   * @notice The order trade parameters that were provided during signature
-   * verification does not match the data stored in this contract _or_ the
-   * AMM has not enabled trading.
+   * @notice Thrown when AppData that was provided during signature verification
+   * does not match the one stored in this contract.
    */
-  error AppDataDoNotMatchHash();
+  error AppDataDoNotMatch();
```
Comment (0xteddybear) [inline src/interfaces/IBCoWPool.sol:57]: `AppDataDoesNotMatch`?

File: test/unit/BCoWPool.t.sol [hunk 4]
```diff
@@ -306,31 +255,28 @@ contract BCoWPool_Unit_IsValidSignature is BaseCoWPoolTest {
...
   }
 
-  modifier _withTradingEnabled(bytes32 _appData) {
-    bCoWPool.set_appDataHash(keccak256(abi.encode(_appData)));
-    _;
-  }
-
-  modifier _withValidCommitment(GPv2Order.Data memory _order) {
+  modifier happyPath(GPv2Order.Data memory _order) {
+    // happy values
```
Comment (0xteddybear) [inline test/unit/BCoWPool.t.sol:259]: nitpick: 
I'd either remove the comments or explain _why_ the value is correct and where is set, something along the lines of 'hard-code it to the same value used when deploying the factory in `setUp` so it passes the first check on `isValidSignature`'

**Review by 0xteddybear — DISMISSED** [review]
only thing blocking merging is the question of wether the appdata should also be immutable in the factory

File: src/contracts/BCoWFactory.sol [hunk 5]
```diff
@@ -13,9 +13,11 @@ import {IBPool} from 'interfaces/IBPool.sol';
  */
 contract BCoWFactory is BFactory {
   address public immutable SOLUTION_SETTLER;
+  bytes32 public immutable APP_DATA;
```
[the remark this answers is not in the document]
Comment (wei3erHase) [reply src/contracts/BCoWFactory.sol:16]: i think that having a `newBPool(bytes32)` is painful, specially when they can redeploy the factory for a given `appData`, perhaps is not supposed to change, but will talk with CoW and double check that

**Review by 0xAustrian — APPROVED** [review]

**Pull request merged**